atlp-rwanda/e-comm-team-amigos-bn
# #184581170 Setup Docker for Backend Application

#### What does this PR do?
Have docker setup for backend API

#### Description of Task to be completed?
Have the following functionality configured

- Running express server using docker compose
- Creating  and running postgres using docker compose

#### How should this be manually tested?
First make sure you have docker installed. In your terminal;

1. Set environment variables. NB: **Set DB_HOST to `db`**
2. Run `docker compose up -d db` to run postgres container
3. Run `docker compose logs` to check if database is ready for connection.
4. Run `docker compose build` to build Docker image.
5. Run `docker compose up app`
6. Run `docker compose exec -it app npx sequelize-cli db:migrate` to create tables
7. Use Postman or any other api testing app to make the following request on 127.0.0.01:3000;
   - `POST /user/create` Body request {userName, email, password}
   - `GET /user/all`

#### What are the relevant pivotal tracker/Trello stories?
Story ID: [#184581170 ](https://www.pivotaltracker.com/story/show/184581170)
